src/taskgraph/generator.py

```diff
@@ -253,6 +264,101 @@ def _load_kinds(self, graph_config, target_kinds=None):
                 except KindNotFound:
                     continue
 
+    def _load_tasks_serial(self, kinds, kind_graph, parameters):
+        all_tasks = {}
+        for kind_name in kind_graph.visit_postorder():
+            logger.debug(f"Loading tasks for kind {kind_name}")
+
+            kind = kinds.get(kind_name)
+            if not kind:
+                message = f'Could not find the kind "{kind_name}"\nAvailable kinds:\n'
+                for k in sorted(kinds):
+                    message += f' - "{k}"\n'
+                raise Exception(message)
+
+            try:
+                new_tasks = kind.load_tasks(
+                    parameters,
+                    {
+                        k: t
+                        for k, t in all_tasks.items()
+                        if t.kind in kind.config.get("kind-dependencies", [])
+                    },
+                    self._write_artifacts,
+                )
+            except Exception:
+                logger.exception(f"Error loading tasks for kind {kind_name}:")
+                raise
+            for task in new_tasks:
+                if task.label in all_tasks:
+                    raise Exception("duplicate tasks with label " + task.label)
+                all_tasks[task.label] = task
+
+        return all_tasks
+
+    def _load_tasks_parallel(self, kinds, kind_graph, parameters):
+        all_tasks = {}
+        futures_to_kind = {}
+        futures = set()
+        edges = set(kind_graph.edges)
+
+        with ProcessPoolExecutor() as executor:
```

Comment:
AIUI we rely on the use of `fork` here, so per the documentation note at https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor should pass
`mp_context=multiprocessing.get_context("fork")` to ensure we get the right thing (and instead of platform == windows we can test for `"fork" in multiprocessing.get_all_start_methods()`?).

I might be confused though because apparently `spawn` is the default on macos already.  Did we test this there?

Reply:
Yup; good callout. Even with forking though, it looks like this crashes on mac repeatedly. https://github.com/python/cpython/issues/77906 suggests that this is a common problem when forking on macOS :(. (`forkserver` also does not work; it fails the same way that `spawn` does.)

Reply:
I ended up turning off support for everything but Linux, to be safe.